pkg/providers/v2/tags.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v2 is an out-of-tree only implementation of the AWS cloud provider.
+// It is not compatible with v1 and should only be used on new clusters.
+package v2
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+const (
+	// TagNameKubernetesClusterPrefix is the tag name we use to differentiate multiple
+	// logically independent clusters running in the same AZ.
+	// we support two tags:
+	// 1) kubernetes.io/cluster/=<clusterName>
+	// 2) for shared resoueces between clusters, kubernetes.io/cluster/<clusterName>=""
+	TagNameKubernetesClusterPrefix = "kubernetes.io/cluster/"
```

Comment:
I think we agreed last time to only use `kubernetes.io/cluster=<cluster-name>` for tagging for now, this means not supporting the "shared" case below.

Reply:
gotcha, removed!